### PR TITLE
Fix wrong PDF page labels (#3836)

### PR DESCRIPTION
**Pull request by gabriel-araujjo**

### CODE REVIEW

File: crates/typst-pdf/src/page.rs [hunk 1]
```diff
@@ -229,17 +229,25 @@ fn write_page(ctx: &mut PdfContext, i: usize, resources_ref: Ref) {
...
+    if !ctx.pages.iter().any(|p| {
+        p.label
+            .as_ref()
+            .is_some_and(|l| l.prefix.is_some() || l.style.is_some())
+    }) {
+        return Vec::new();
+    }
+
+    // If there pages with empty labels between labeled pages, we must write
+    // empty PageLabel entries.
```
Comment (laurmaedje) [inline crates/typst-pdf/src/page.rs:242]:
```suggestion
    // If there are pages with empty labels between labeled pages, we must write
    // empty PageLabel entries.
```
and should be moved below directly above the `page.label.as_ref().unwrap_or(&empty_label);` line

Comment (gabriel-araujjo) [reply crates/typst-pdf/src/page.rs]: I applied the suggestion on commit  3f983fd0a27d8ee7c3687ee9c27a6771c61353b2.

**Comment by laurmaedje** [comment]
Thanks!

**Comment by istudyatuni** [comment]
It seems that #3836 should be closed

**Pull request merged**